crates/valence_boss_bar/src/lib.rs

```diff
@@ -0,0 +1,191 @@
+#![allow(clippy::type_complexity)]
+
+use bevy_app::Plugin;
+use bevy_ecs::query::{Added, Changed};
+use bevy_ecs::removal_detection::RemovedComponents;
+use bevy_ecs::system::Query;
+use components::{BossBarFlags, BossBarHealth, BossBarStyle, BossBarTitle, BossBarViewers};
+use packet::{BossBarAction, BossBarS2c};
+use valence_client::Client;
+use valence_core::despawn::Despawned;
+use valence_core::protocol::encode::WritePacket;
+use valence_core::uuid::UniqueId;
+
+pub mod components;
+pub mod packet;
+
+pub struct BossBarPlugin;
+
+impl Plugin for BossBarPlugin {
+    fn build(&self, app: &mut bevy_app::App) {
+        app.add_system(handle_boss_bar_title_update)
+            .add_system(handle_boss_bar_health_update)
+            .add_system(handle_boss_bar_style_update)
+            .add_system(handle_boss_bar_flags_update)
+            .add_system(handle_boss_bar_viewers_update)
+            .add_system(handle_boss_bar_despawn)
+            .add_system(handle_client_disconnection);
+    }
+}
+
+/// System that sends a bossbar update title packet to all viewers of a boss bar
+/// that has had its title updated.
+fn handle_boss_bar_title_update(
+    mut boss_bars: Query<(&UniqueId, &BossBarTitle, &mut BossBarViewers), Changed<BossBarTitle>>,
```

Comment:
Does not need to be mutable. (issue repeated in other systems)

```suggestion
    mut boss_bars: Query<(&UniqueId, &BossBarTitle, &BossBarViewers), Changed<BossBarTitle>>,
```